assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { store as interactivityApiStore } from '@woocommerce/interactivity';
+
+interface State {
+	[ key: string ]: unknown;
+}
+
+interface Context {
+	productGallery: { numberOfThumbnails: number };
+}
+
+interface Selectors {
+	productGallery: {
+		getNumberOfPages: ( store: unknown ) => number;
+	};
+}
+
+interface Store {
+	state: State;
+	context: Context;
+	selectors: Selectors;
+	ref: HTMLElement;
+}
+
+type SelectorsStore = Pick< Store, 'context' | 'selectors' >;
+
+interactivityApiStore( {
+	selectors: {
+		productGallery: {
+			getNumberOfPages: ( store: SelectorsStore ) => {
```

Comment:
The selectors should not have the prefix `get`. You can read more about here: https://github.com/WordPress/gutenberg/discussions/53586. For the `ProductButton`, the selectors don't have the `get` prefix.